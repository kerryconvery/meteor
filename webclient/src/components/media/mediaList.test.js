import React from 'react';
import { shallow } from 'enzyme';
import MediaList from './mediaList';


describe('MediaList', () => {
  it('should call onItemClicked', () => {
    const media = [
      {
        name: 'media1',
        isDirector: false,
        uri: '',
      },
    ];
    const mockOnClick = jest.fn();
    const wrapper = shallow(<MediaList items={media} profile='profile1' onItemClicked={mockOnClick} />);

    wrapper.find('ListGroupItem').simulate('click');

    expect(mockOnClick).toHaveBeenCalled();
  });
});

test('renders a list of media', () => {
  const media = [
    {
      name: 'media1',
      isDirectory: false,
      uri: '',
    },
    {
      name: 'media2',
      isDirectory: true,
      uri: '',
    },
  ];
  const wrapper = shallow(<MediaList items={media} profile='profile1' onItemClicked={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

