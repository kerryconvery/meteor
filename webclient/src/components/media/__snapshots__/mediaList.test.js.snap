// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of media 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MediaList
    items={
      Array [
        Object {
          "isDirector": false,
          "name": "media1",
          "uri": "",
        },
        Object {
          "isDirector": true,
          "name": "media2",
          "uri": "",
        },
      ]
    }
    onItemClicked={[Function]}
    profile="profile1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ListGroupItem
          action={true}
          onClick={[Function]}
          tag="button"
        >
          <img
            alt=""
            src="/api/profiles/profile1/media/thumbnail?uri="
          />
          <span
            style={
              Object {
                "marginLeft": "5px",
              }
            }
          >
            media1
          </span>
        </ListGroupItem>,
        <ListGroupItem
          action={true}
          onClick={[Function]}
          tag="button"
        >
          <img
            alt=""
            src="/api/profiles/profile1/media/thumbnail?uri="
          />
          <span
            style={
              Object {
                "marginLeft": "5px",
              }
            }
          >
            media2
          </span>
        </ListGroupItem>,
      ],
      "tag": "ul",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "media1",
        "nodeType": "function",
        "props": Object {
          "action": true,
          "children": Array [
            <img
              alt=""
              src="/api/profiles/profile1/media/thumbnail?uri="
            />,
            <span
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
            >
              media1
            </span>,
          ],
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "/api/profiles/profile1/media/thumbnail?uri=",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "media1",
              "style": Object {
                "marginLeft": "5px",
              },
            },
            "ref": null,
            "rendered": "media1",
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "media2",
        "nodeType": "function",
        "props": Object {
          "action": true,
          "children": Array [
            <img
              alt=""
              src="/api/profiles/profile1/media/thumbnail?uri="
            />,
            <span
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
            >
              media2
            </span>,
          ],
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "/api/profiles/profile1/media/thumbnail?uri=",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "media2",
              "style": Object {
                "marginLeft": "5px",
              },
            },
            "ref": null,
            "rendered": "media2",
            "type": "span",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListGroupItem
            action={true}
            onClick={[Function]}
            tag="button"
          >
            <img
              alt=""
              src="/api/profiles/profile1/media/thumbnail?uri="
            />
            <span
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
            >
              media1
            </span>
          </ListGroupItem>,
          <ListGroupItem
            action={true}
            onClick={[Function]}
            tag="button"
          >
            <img
              alt=""
              src="/api/profiles/profile1/media/thumbnail?uri="
            />
            <span
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
            >
              media2
            </span>
          </ListGroupItem>,
        ],
        "tag": "ul",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "media1",
          "nodeType": "function",
          "props": Object {
            "action": true,
            "children": Array [
              <img
                alt=""
                src="/api/profiles/profile1/media/thumbnail?uri="
              />,
              <span
                style={
                  Object {
                    "marginLeft": "5px",
                  }
                }
              >
                media1
              </span>,
            ],
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "/api/profiles/profile1/media/thumbnail?uri=",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "media1",
                "style": Object {
                  "marginLeft": "5px",
                },
              },
              "ref": null,
              "rendered": "media1",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "media2",
          "nodeType": "function",
          "props": Object {
            "action": true,
            "children": Array [
              <img
                alt=""
                src="/api/profiles/profile1/media/thumbnail?uri="
              />,
              <span
                style={
                  Object {
                    "marginLeft": "5px",
                  }
                }
              >
                media2
              </span>,
            ],
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "/api/profiles/profile1/media/thumbnail?uri=",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "media2",
                "style": Object {
                  "marginLeft": "5px",
                },
              },
              "ref": null,
              "rendered": "media2",
              "type": "span",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
